be/src/olap/cumulative_compaction_policy.cpp

```diff
@@ -460,16 +461,11 @@ void NumBasedCumulativeCompactionPolicy::calculate_cumulative_point(
 }
 
 void CumulativeCompactionPolicy::pick_candidate_rowsets(
-        int64_t skip_window_sec,
         const std::unordered_map<Version, RowsetSharedPtr, HashOfVersion>& rs_version_map,
         int64_t cumulative_point, std::vector<RowsetSharedPtr>* candidate_rowsets) {
-    int64_t now = UnixSeconds();
     for (auto& it : rs_version_map) {
         // find all rowset version greater than cumulative_point and skip the create time in skip_window_sec
-        if (it.first.first >= cumulative_point &&
-            ((it.second->creation_time() + skip_window_sec < now)
-             // this case means a rowset has been compacted before which is not a new published rowset, so it should participate compaction
-             || (it.first.first != it.first.second))) {
```

Comment:
We should not remove this condition?

Reply:
((it.second->creation_time() + skip_window_sec < now)
             // this case means a rowset has been compacted before which is not a new published rowset, so it should participate compaction
             || (it.first.first != it.first.second)))
is useless here, if no skip_window condition, then (it.first.first != it.first.second) is also unnecessary

Reply:
Ah... you're right.